hal/src/nRF52840/ble_hal_impl.h

```diff
@@ -15,32 +15,148 @@
  * License along with this library; if not, see <http://www.gnu.org/licenses/>.
  */
 
-#pragma once
+#ifndef BLE_HAL_IMPL_H
+#define BLE_HAL_IMPL_H
 
-#include "ble_types.h" // For BLE_CONN_HANDLE_INVALID
-#include "ble_gatt.h"
 
-#include "sdk_config.h"
+#if HAL_PLATFORM_BLE
 
-#if !defined(NRF_SDH_BLE_GATT_MAX_MTU_SIZE) || (NRF_SDH_BLE_GATT_MAX_MTU_SIZE == 0)
-#error "NRF_SDH_BLE_GATT_MAX_MTU_SIZE is not defined"
-#endif
+#include "ble_gap.h"
+#include "ble_gatts.h"
+#include "sdk_config_system.h"
 
-// Invalid connection handle
-// TODO: Prefix all BLE HAL definitions with "hal_"
-#define BLE_INVALID_CONN_HANDLE BLE_CONN_HANDLE_INVALID
 
-// Invalid attribute handle
-#define BLE_INVALID_ATTR_HANDLE BLE_GATT_HANDLE_INVALID
+/**< Number of microseconds in 0.625 milliseconds. */
+#define BLE_UNIT_0_625_MS                           625
+/**< Number of microseconds in 1.25 milliseconds. */
+#define BLE_UNIT_1_25_MS                            1250
+/**< Number of microseconds in 10 milliseconds. */
+#define BLE_UNIT_10_MS                              10000
 
-// Maximum number of peripheral connections
-#define BLE_MAX_PERIPH_CONN_COUNT NRF_SDH_BLE_PERIPHERAL_LINK_COUNT
+#define BLE_MSEC_TO_UNITS(TIME, RESOLUTION)         (((TIME) * 1000) / (RESOLUTION))
 
-// Maximum number of services per profile
-#define BLE_MAX_SERVICE_COUNT 1
+#define BLE_ROLE_INVALID                            BLE_GAP_ROLE_INVALID
```

Comment:
These should probably not be so directly exposed. The API should present its own `ble_role` enum and the mapping between the implementation internal type and the API `ble_role` should be hidden in the implementation itself.